server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -279,6 +279,9 @@ final ObjectMapper getDynamicObjectMapper(String name) {
      * Add a new runtime field dynamically created while parsing.
      */
     public final void addDynamicRuntimeField(RuntimeField runtimeField) {
+        if (newFieldsSeen.add(runtimeField.name())) {
```

Comment:
A runtime field can have the same name as an indexed field, in which case it will shadow the indexed field at search time. I wondered if it's ok to have the same set for both runtime fields and indexed fields and it does look ok because when it comes to dynamic mappings, when we encounter a new field, either we map it as a runtime field or as an indexed field, and never both, so there are never going to be collisions. Shall we add a comment to clarify that?

Reply:
@javanna Good suggestion,  addressed in dbf20223bce7a5340f803eb59188a9d7d5aa5418

Indeed for dynamic updates, a field can be mapped either as runtime field or as indexed field, so no problem here.

For PUT _mapping updates, a user can introduce a new  runtime field with the same name as existing indexed field. In this case, this new runtime field will be added to the total count, and can cause an exception for total field counts.